pydantic_ai_slim/pydantic_ai/messages.py

```diff
@@ -122,30 +114,42 @@ class RetryPrompt:
     timestamp: datetime = field(default_factory=_now_utc)
     """The timestamp, when the retry was triggered."""
 
-    role: Literal['user'] = 'user'
-    """Message source identifier, this type is available on all messages as either 'user' or 'model'."""
-
-    message_kind: Literal['retry-prompt'] = 'retry-prompt'
-    """Message type identifier, this type is available on all messages as a discriminator."""
+    kind: Literal['retry-prompt'] = 'retry-prompt'
+    """Part type identifier."""
 
     def model_response(self) -> str:
         if isinstance(self.content, str):
             description = self.content
         else:
-            json_errors = ErrorDetailsTa.dump_json(self.content, exclude={'__all__': {'ctx'}}, indent=2)
+            json_errors = error_details_ta.dump_json(self.content, exclude={'__all__': {'ctx'}}, indent=2)
             description = f'{len(self.content)} validation errors: {json_errors.decode()}'
         return f'{description}\n\nFix the errors and try again.'
 
 
+UserMessagePart = Annotated[Union[SystemPrompt, UserPrompt, ToolReturn, RetryPrompt], pydantic.Discriminator('kind')]
```

Comment:
This is somewhat inconsistent - we should probably use `SystemPromptPart`, `UserPromptPart`, etc to be consistent with the `TextPart` and `ToolCallPart` below

Reply:
I'll note, I don't really like `SystemPromptPart`, it's kind of verbose, but I do think we should be consistent across `UserMessagePart` and `ModelMessagePart`